website/source/docs/guides/consul-template.html.md

```diff
@@ -0,0 +1,178 @@
+to install it in a way that meets their use case or use cases. This can mean 
+installing a few instances for the entire cluster or dozens on a single host.
+
+## Install Consul Template
+
+For this guide, we are using a local Consul agent in development
+mode which can be started with `consul agent -dev`. To quickly set 
+up a local Consul agent, refer to the getting started [guide](https://learn.hashicorp.com/consul/getting-started/install). The 
+Consul agent must be running to complete all of the following
+steps. 
+
+The Consul template tool is not included with the Consul binary and will
+need to be installed separately. It can be installed from a precompiled 
+binary or compiled from source. We will be installing the precompiled binary.
+
+First, download the binary from the [Consul Template releases page](https://releases.hashicorp.com/consul-template/).
+
+```sh
+curl -O https://releases.hashicorp.com/consul-template/0.19.5/consul-template<_version_OS>.tgz
+```
+
+Next, extract the binary and move it into you `$PATH`.
+
+```sh
+tar -zxf consul-template<_version_OS>.tgz
+``` 
+
+To compile from source, please see the instructions in the
+[contributing section in Github](https://github.com/hashicorp/consul-template#contributing).
+
+## Use Case: Consul KV
+
+In this first use case example, we will create a simple template that contains the HashiCorp
+address, run `consul-template`, add a value to Consul KV for HashiCorp's address, and 
+finally view the rendered file.   
+
+First, we will need to create a template file `find_address.tpl` to query 
+Consul's KV store:
+
+```liquid
+{{ key "/hashicorp/street_address" }}
+```
+
+Next, we will run `consul-template` specifying both
+the template to use and the file to update. 
+
+```shell
+$ consul-template -template "find_address.tpl:hashicorp_address.txt" 
+```
+
+The `consul-template` process will continue to run until you kill it with `CRTL+c`. 
+For now, we will leave it running. 
+
+Finally, open a new terminal so we can write data to the key in Consul using the command 
+line interface.
+
+```shell
+$ consul kv put hashicorp/street_address "101 2nd St"
+$ Success! Data written to: hashicorp/street_address
+```
+
+We can ensure the data was written by viewing the `hashicorp_address.txt`
+file which will be located in the same directory where `consul-template`
+was run.
+
+```shell
+$ cat hashicorp_address.txt
+101 2nd St
+```
+
+If you update the key `hashicorp/street_address`, you can see the changes to the file 
+immediately. Go ahead and try `consul kv put hashicorp/street_address "201 Stevenson ST"`.
+
+You can see that this simple process can have powerful implications. For example, it is
+possible to use this same process for updating your [HAProxy load balancer 
+configuration](https://github.com/hashicorp/consul-template/blob/master/examples/haproxy.md). 
+
+You can now kill the `consul-template` process with `CTRL+c`. 
```

Comment:
This is fantastic!

Do you think we can include a step of setting a command to run on update since in practice almost all systems that need config files written on disk also need HUP to reload them.

One option for this example you could make the command be something like `cat hashicorp_address.txt >> /tmp/address_history.txt` (append the new contents to a temp "log" file). And then the user can observe this happening by running `tail -f /tmp/address_history.txt` in another terminal - then it highlights the real-time nature too.

It's a little arcane vs HUP so would be good to spell out that as a more natural use-case but I think it would be an interesting demo and get people clear on the fact that we don't just write files but can trigger things too?